### PR TITLE
Fix edge cases in PostgreSQL maintenance

### PR DESCRIPTION
**Pull request by glrf**
This commit fixes two separate issue with the PostreSQL maintenance script. Both are related to waiting for the upgrade to complete.

* Fixes a panic in case we receive an event with an SGDbOps without any conditions
* Fixes an issue where the script will run repack prematurely, in case there is an event on a different, completed, SGDbOps in the namespace.

The second point is fairly theoretical, but the second one can be reproduced locally, even though it does not seem to occur in production. Probably a timing issue.

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.

### CODE REVIEW

File: pkg/maintenance/postgresql.go [hunk 1]
```diff
@@ -168,9 +168,12 @@ func (p *PostgreSQL) waitForUpgrade(ctx context.Context) error {
 			switch event.Type {
 			case watch.Modified:
 				ops, _ := event.Object.(*stackgresv1.SGDbOps)
+				if ops.Status.Conditions == nil || ops.Spec.Op != string(op) {
```
Comment (zugao) [inline pkg/maintenance/postgresql.go:171]: I think there's no status whatsoever for non recycled CRD. Therefore you might want to check `status` for nil?

Comment (glrf) [reply pkg/maintenance/postgresql.go:171]: Yeah but the status isn't a pointer :)

Comment (zugao) [reply pkg/maintenance/postgresql.go:171]: Then no problem :D

**Review by zugao — APPROVED** [review]
LGTM

**Pull request merged**